File: app/src/main/java/com/example/merchio/adapters/CheckoutAdapter.java
package com.example.merchio.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.merchio.R;
import com.example.merchio.models.CartItem;

import java.text.NumberFormat;
import java.util.List;
import java.util.Locale;

public class CheckoutAdapter
        extends RecyclerView.Adapter<CheckoutAdapter.ViewHolder> {

    Context context;
    List<CartItem> itemList;

    public CheckoutAdapter(Context context,
                           List<CartItem> itemList) {

        this.context = context;
        this.itemList = itemList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(
            @NonNull ViewGroup parent,
            int viewType) {

        View view =
                LayoutInflater.from(context)
                        .inflate(
                                R.layout.item_checkout,
                                parent,
                                false
                        );

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(
            @NonNull ViewHolder holder,
            int position) {

        CartItem item = itemList.get(position);

        holder.txtName.setText(item.getProductName());

        holder.txtType.setText(item.getType());

        holder.txtQty.setText(
                "Qty: " + item.getQuantity()
        );

        int totalPrice =
                item.getProductPrice()
                        * item.getQuantity();

        holder.txtPrice.setText(
                rupiah(totalPrice)
        );

        Glide.with(context)
                .load(item.getProductImage())
                .into(holder.imgProduct);
    }

    @Override
    public int getItemCount() {
        return itemList.size();
    }

    private String rupiah(int amount) {

        NumberFormat format =
                NumberFormat.getCurrencyInstance(
                        new Locale("id","ID")
                );

        return format.format(amount)
                .replace(",00","");
    }

    public static class ViewHolder
            extends RecyclerView.ViewHolder {

        ImageView imgProduct;

        TextView txtName,
                txtType,
                txtQty,
                txtPrice;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            imgProduct =
                    itemView.findViewById(R.id.imgProduct);

            txtName =
                    itemView.findViewById(R.id.txtName);

            txtType =
                    itemView.findViewById(R.id.txtType);

            txtQty =
                    itemView.findViewById(R.id.txtQty);

            txtPrice =
                    itemView.findViewById(R.id.txtPrice);
        }
    }
}

File: app/src/main/java/com/example/merchio/adapters/CheckoutAdapter.kt
package com.example.merchio.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.merchio.R
import com.example.merchio.data.model.CartDisplayItem
import com.example.merchio.utils.CurrencyFormatter
import com.example.merchio.utils.loadMerchioImage

class CheckoutAdapter : RecyclerView.Adapter<CheckoutAdapter.CheckoutViewHolder>() {
    private val items = mutableListOf<CartDisplayItem>()

    fun submitList(data: List<CartDisplayItem>) {
        items.clear()
        items.addAll(data)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CheckoutViewHolder {
        return CheckoutViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_checkout, parent, false))
    }

    override fun onBindViewHolder(holder: CheckoutViewHolder, position: Int) = holder.bind(items[position])
    override fun getItemCount(): Int = items.size

    inner class CheckoutViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val image: ImageView = itemView.findViewById(R.id.imgProduct)
        private val name: TextView = itemView.findViewById(R.id.txtName)
        private val type: TextView = itemView.findViewById(R.id.txtType)
        private val qty: TextView = itemView.findViewById(R.id.txtQty)
        private val price: TextView = itemView.findViewById(R.id.txtPrice)
        fun bind(item: CartDisplayItem) {
            image.loadMerchioImage(item.product.imageUrl)
            name.text = item.product.name
            type.text = item.cart.variantLabel
            qty.text = "Qty ${item.cart.quantity}"
            price.text = CurrencyFormatter.rupiah(item.lineTotal)
        }
    }
}
